src/components/Pages/SearchPage/SearchPage.jsx

```diff
@@ -19,17 +21,43 @@ const mapStateToProps = state => ({
   sortingType: state.sortingType,
   searchOption: state.searchOption,
   isSearching: state.isSearching,
-  error: state.isSearching
+  error: state.isSearching,
+  searchValue: state.searchValue
 });
 
 export class SearchPage extends Component {
+  state = {};
+
+  static getDerivedStateFromProps(props, state) {
+    const { term } = props.match.params; // receiving current search term from URL
+    const { searchValue } = props; // receiving stored search term
+    const {
+      sortingType,
+      searchOption,
+      getSearchData,
+      updateSearchValue
+    } = props;
+
+    if (term && term !== searchValue) {
+      // checking whether we should perform new search or not
+      getSearchData(term, sortingTypeForSearch[sortingType], searchOption);
+      updateSearchValue(term);
+      history.push(`/search/${term}`);
```

Comment:
term comes from url, so there is no need to put it there once again )
in some case it leads to issue with read-only input and fetching data on each input key: try to open movie details, then open `/search/hello` and try to enter something in your input.